src/bika/lims/content/abstractanalysis.py

```diff
@@ -685,9 +685,14 @@ def getEarliness(self):
         :rtype: int
         """
         maxtime = self.getMaxTimeAllowed()
+        is_zero_or_none = ("0", 0, None)
         if not maxtime:
             # No Turnaround time is set for this analysis
             return 0
+        if maxtime.get("hours") in is_zero_or_none \
+            and maxtime.get("minutes") in is_zero_or_none \
+            and maxtime.get("days") in is_zero_or_none:
+            return 0
```

Comment:
Way easier if you check if the result of `api.to_minutes(**maxtime)` is 0 instead